### PR TITLE
[l10n] Improve Persian (fa-IR) locale

### PR DESCRIPTION
**Pull request by amiryxe**
fix some spaces and change some words

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/x-data-grid/src/locales/faIR.ts [hunk 1]
```diff
@@ -110,14 +110,14 @@ const faIRGrid: Partial<GridLocaleText> = {
   columnMenuFilter: 'فیلتر',
   columnMenuHideColumn: 'مخفی',
   columnMenuUnsort: 'نامرتب‌کردن',
-  columnMenuSortAsc: 'مرتب‌کردن صعودی',
-  columnMenuSortDesc: 'مرتب‌کردن نزولی',
+  columnMenuSortAsc: 'مرتب‌سازی صعودی',
+  columnMenuSortDesc: 'مرتب‌سازی نزولی',
```
Comment (oliviertassinari) [inline packages/x-data-grid/src/locales/faIR.ts:114]: Are we sure about this? It should describe an action. Google translate says it's passive voice now.

Comment (amiryxe) [reply packages/x-data-grid/src/locales/faIR.ts:114]: > Are we sure about this? It should describe an action. Google translate says it's passive voice now.

Both are correct, but in this case, the term `مرتب‌سازی` is mostly used in Persian.

Comment (amiryxe) [reply packages/x-data-grid/src/locales/faIR.ts:114]: Thank you @MBilalShafi for following up. This case is not very controversial and as I said, both cases are correct and if you are more comfortable with the first case, there is no problem. However, as a Persian speaker, I asked my other friends about this and they agreed with me.

Comment (oliviertassinari) [reply packages/x-data-grid/src/locales/faIR.ts:114]: Kendo UI jQuery data grid translations: https://kendo.cdn.telerik.com/2024.2.514/js/messages/kendo.messages.fa-IR.min.js

```
sortAscending:"مرتب سازی صعودی",
sortDescending:"مرتب سازی نزولی",
```

This looks close to @amiryxe's proposal.

Comment (MBilalShafi) [reply packages/x-data-grid/src/locales/faIR.ts:114]: Alright then, let's merge it in that case.
Thank you for your contribution. 🙏

**Review by MBilalShafi — APPROVED** [review]

**Pull request merged**